chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/MeaningfulUseController.java

```diff
@@ -0,0 +1,185 @@
+						muu.setError("Line " + muu.getCsvLineNumber() + ": Field \"num_meaningful_use\" with value \"" + currRecord.get(1).trim() + "\" is invalid. "
+								+ "Value in field \"num_meaningful_use\" must be an integer.");
+						muusToUpdate.add(muu);
+						uniqueMuusFromFile.add(muu.getProductNumber());
+					}
+					catch (IOException e){
+						muu.setProductNumber(chplProductNumber);
+						muu.setCsvLineNumber(i);
+						Integer dupLineNumber = null;
+						// get line number with duplicate chpl_product_number
+						for (MeaningfulUseUser entry: muusToUpdate) {
+						     if (entry.getProductNumber().equals(muu.getProductNumber())){
+						    	 dupLineNumber = entry.getCsvLineNumber();
+						     }
+						   }
+						muu.setError("Line " + muu.getCsvLineNumber() + ": Field \"chpl_product_number\" with value \"" + muu.getProductNumber() + "\" is invalid. "
+								+ "Duplicate \"chpl_product_number\" at line " + dupLineNumber);
+						muusToUpdate.add(muu);
+					}
+				}
+			}
+		} catch(IOException ioEx) {
+			logger.error("Could not get input stream for uploaded file " + file.getName());			
+			throw new ValidationException("Could not get input stream for uploaded file " + file.getName());
+		} finally {
+			 try { parser.close(); } catch(Exception ignore) {}
+			try { reader.close(); } catch(Exception ignore) {}
+		}
+		
+		try {
+			meaningfulUseUserResults = cpManager.updateMeaningfulUseUsers(muusToUpdate);
+		} catch (JsonProcessingException e) {
+			e.printStackTrace();
+		} catch (EntityCreationException e) {
+			e.printStackTrace();
+		} catch (EntityRetrievalException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		} 
+		
+		return meaningfulUseUserResults;
+	}
+	
+	@ApiOperation(value="Get a single date value to indicate when the last meaningful use user file data is accurate as of.", 
+			notes="This is a single system-wide value.")
+	@RequestMapping(value="/accurate_as_of", method=RequestMethod.GET,
+			produces="application/json; charset=utf-8") 
+	public @ResponseBody MeaningfulUseAccurateAsOf getAccurateAsOfDate() {
+		MeaningfulUseAccurateAsOfDTO dto = muManager.getMeaningfulUseAccurateAsOf();
+		MeaningfulUseAccurateAsOf muuAccurate = new MeaningfulUseAccurateAsOf(dto);
+		if(StringUtils.isEmpty(muuAccurate.getId().toString()) && StringUtils.isEmpty(muuAccurate.getAccurateAsOfDate().toString())){
+			muuAccurate.getErrorMessages().add("Could not obtain the accurate as of date.");
+		}
+		return muuAccurate;
+	}
+	
+	@ApiOperation(value="Update the Meaningful Use Accurate As Of date.", 
+			notes="Accurate as of date value can be edited by a user with ROLE_ADMIN and ROLE_CMS_STAFF.")
+	@RequestMapping(value="/accurate_as_of", method=RequestMethod.POST,
+			produces="application/json; charset=utf-8") 
+	public @ResponseBody MeaningfulUseAccurateAsOf updateMeaningfulUseAccurateAsOf(@RequestParam(required=true) MeaningfulUseAccurateAsOf meaningfulUseAccurateAsOf) {
```

Comment:
In terms of RESTful API design, I wouldn't expect a user to have to send in this whole object when really all they want to set is a single date. Could you just let them pass in a date? I think all the other info (last modified user, etc) is already available to you in the UI. I thought we originally said that the table will always have a single row so I don't think you need the id passed in to update it. I could have missed some discussion so if there is reason for doing it this way, that's fine.